### ISSUE TITLE
New Lint: Don't modify loop condition variables more than once per iteration

### ISSUE THREAD

**Issue by Evrey** [Manishearth/rust-clippy#1679]
Based on [this HIC++ guideline](http://www.codingstandard.com/rule/6-2-3-do-not-alter-a-control-or-counter-variable-more-than-once-in-a-loop/). To give you a short example of what would be linted:

```Rust
while do_it != DoNothing {
    do_it = do_wörk();
    if special_case { do_it = magic(); } // WARNING: dude, you already modified `do_it`!
}
```

The idea is that for all mutable variables that are checked in a `while` loop's condition, this variable should only be modified once per iteration to simplify reasoning about the loop's control flow.
